main.go

```diff
@@ -60,78 +69,118 @@ func readCircleYaml() (models.CircleYamlV1, error) {
 	return out, nil
 }
 
+// convertToV2 uses the CircleCI 1.0 formatted YAML
+// and other cues in the repo (Makefile, presence of swagger.yml, etc)
+// to create CircleCI 2.0 formatted YAML
 func convertToV2(v1 models.CircleYamlV1) (models.CircleYamlV2, error) {
 	v2 := models.CircleYamlV2{
 		Version: 2,
 	}
 
-	// TODO: dynamically determine org/repo
-	org := "Clever"
-	repo := "kinesis-to-firehose"
-
-	// TODO: Determine working dir depending on language
-	v2.Jobs.Build.WorkingDirectory = fmt.Sprintf("/go/src/github.com/%s/%s", org, repo)
-	v2.Jobs.Build.Docker = []models.Docker{
-		models.Docker{
-			Image: "circleci/golang:1.8",
+	// Determine base image to use based on app type (go/wag/node/...) and language version
+	imageConstraints := determineImageConstraints()
+	appType := imageConstraints.AppType
+	primaryImage := getImage(imageConstraints)
+	v2.Jobs.Build.Docker = []models.DockerImage{
+		models.DockerImage{
+			Image: primaryImage,
 		},
 	}
+	// @TODO (INFRA-3159): Determine and add additional database image(s) needed
```

Comment:
hopefully grepping on `MONGO_URL` is enough for at least the mongo usage?

Reply:
🤞